### ISSUE TITLE
[Bug]: Crash when loading pages with Electron 13

### ISSUE THREAD

**Issue by pushkin-** [electron/electron#29160]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

13.0.0 beta 26

### What operating system are you using?

Windows

### Operating System Version

Windows 10

### What arch are you using?

x64

### Last Known Working Electron version

12

### Expected Behavior

I expect to be able to load a page without a crash

### Actual Behavior

When loading a simple page into a BrowserWindow, devtools shows:

>VM95 worker.js:82 Uncaught ReferenceError: SharedArrayBuffer is not defined
    at VM95 worker.js:82
    at NativeModule.compileForInternalLoader (VM10 loaders.js:283)
    at nativeModuleRequire (VM10 loaders.js:312)
    at VM94 worker_threads.js:9
    at NativeModule.compileForInternalLoader (VM10 loaders.js:283)
    at nativeModuleRequire (VM10 loaders.js:312)
    at VM93 inspector.js:30
    at NativeModule.compileForInternalLoader (VM10 loaders.js:283)
    at NativeModule.compileForPublicLoader (VM10 loaders.js:225)
    at loadNativeModule (VM50 helpers.js:35)

### Additional Information

1. start main.js and look at devtools:

main.js:
```
const { app, BrowserWindow } = require("electron");

(async () =>
{
	await app.whenReady();
	const win = new BrowserWindow();
	await win.loadURL("https://example.com");
	win.webContents.openDevTools();
})();
```

In my actual app, my preload script appears to not load at all. Possibly related. Similar issue [here](https://github.com/electron/electron/issues/28865).

In my case though it's actually crashing, which it doesn't do in Electron 12.

**Comment by VerteDinde** [comment]
Hey @pushkin-, thanks for reporting! I'm not able to repro this on my Windows machine, could you take a look at my gist and let me know if there's anything I'm missing? https://gist.github.com/77099b7e58fdb29c67d1562e1ef3c04c If you have another repo where this is happening, happy to take a look.

**Comment by pushkin-** [comment]
Weird - I can still repro with your gist. (rather the gist is the same as mine just with extra unused files, so I just got rid of them)

What Windows version are you on? I'm on Version 20H2 (OS Build 19042.928) if it makes a difference.

**Comment by pushkin-** [comment]
But I can't repro on my personal laptop with beta 27. Windows 10 Version 2004 (OS build 19041.985)

**Comment by be-dcodeit** [comment]
We are also getting crashes on Electron 13, however only on 11th generation Intel CPUs (which support Intels new CET security feature) https://www.offensive-security.com/offsec/intel-cet-in-action/#fn:3

When debugging with WinDbg we can do see that the CET shadow stack exception is thrown which crashes the process (see error message attached)
![image](https://user-images.githubusercontent.com/4749907/119541020-63110e80-bd8e-11eb-9952-cc9a7536624f.png)
Chrome introduced support for CET in version 90, it appears this causes issues?

**Comment by VerteDinde** [comment]
Hey @pushkin- and @be-dcodeit: we did some more hunting on this and we think this is almost definitely is related to CET. 

Quick summary of what's happening: for Chromium, v8 de-optimisation prevents use of CET in the renderer process. However, in Electron because we also use v8 in our main process. This CET conflict with v8 in the main process is causing the hard crashes we're seeing on certain types of newer hardware. We're shipping a fix now in the newest version of 13, but would appreciate you both testing and letting us know if it fixes the issue for you for sure.

Thanks so much for reporting this! 🙇‍♀️

**Comment by VerteDinde** [comment]
@pushkin- @be-dcodeit 13.1.1 was just released - could you let me know if this fixes the crashes for you? I’ll keep this issue open until we confirm, and will close it out after if 13.1.1 has the fix. 🙂

**Comment by pushkin-** [comment]
@VerteDinde Sweet - I confirmed that it fixes the issue. Thanks!

**Comment by be-dcodeit** [comment]
@VerteDinde Crash seems to be resolved in 13.1.1, thanks.
